train.py

```diff
@@ -1,216 +1,228 @@
-        learning_rate = 0.0005 
+    Modes:
+        Speed training (default)
+        Accuracy training
+        Extended training (set max_epochs as desired, for example 50+)
+    """
+    print(f"Training started")
+
+    timestamp = datetime.datetime.now().strftime("%Y-%b-%d-%H%M%S").lower()
+    name, in_files, out_files = IO.validate_source(source_dir)
+    out_dir = f'{name}_{timestamp}'
+    os.makedirs(f'models/{out_dir}')
+    test_size = test_pct/100
+
+    if training_mode == 'extended':
+         # Extended Training (~60x longer than Accuracy Training)
+        learning_rate = 0.0005
         conv1d_strides = 3
         conv1d_filters = 36
         hidden_units= 96
-
-    # Create Sequential Model ###########################################
+    elif training_mode == 'accurate':
+        # ~10x longer than Speed Training
+        learning_rate = 0.01
+        conv1d_strides = 4
+        conv1d_filters = 36
+        hidden_units= 64
+    else:
+        learning_rate = 0.01
+        conv1d_strides = 12
+        conv1d_filters = 16
+        hidden_units = 36
+    print(f'Params set for training mode "{training_mode}"')
+
+    # =========================================================================
+    # Create Sequential Model
+    # -------------------------------------------------------------------------
+    # A Sqeuential model is a linear stack of layers. "Same" padding is used to
+    # ensure the output is the same size as the input. The input shape is the
+    # number of samples (input_size) and the number of features (1). The model
+    # is compiled with the Adam optimizer and the custom loss function "esr".
+    #
+    # The history object is used to store the results of model.fit and append
+    # metadata about the training process to the saved model file.
+    # =========================================================================
     clear_session()
+    history = None
     model = Sequential()
-    model.add(Conv1D(conv1d_filters, 12,strides=conv1d_strides, activation=None, padding='same',input_shape=(input_size,1)))
-    model.add(Conv1D(conv1d_filters, 12,strides=conv1d_strides, activation=None, padding='same'))
+    model.add(Input(shape=(input_size, 1)))
+    model.add(Conv1D(conv1d_filters, 12, strides=conv1d_strides, activation=None, padding='same'))
+    model.add(Conv1D(conv1d_filters, 12, strides=conv1d_strides, activation=None, padding='same'))
     model.add(LSTM(hidden_units))
     model.add(Dense(1, activation=None))
-    model.compile(optimizer=Adam(learning_rate=learning_rate), loss=error_to_signal, metrics=[error_to_signal])
-    print(model.summary())
-
-    # Load and Preprocess Data ###########################################
-    in_rate, in_data = wavfile.read(args.in_file)
-    out_rate, out_data = wavfile.read(args.out_file)
+    model.compile(optimizer=Adam(learning_rate=learning_rate), loss=Data.esr, metrics=[Data.esr])
+    print('\n') ;model.summary(); print('\n')
```

Comment:
model.summary() does it's own printing, so this double prints and appends a stray "None" after the summary which was confusing. Dropped and added new line on either side.